clippy_lints/src/utils/format_args_collector.rs

```diff
@@ -5,27 +5,20 @@ use rustc_ast::{Crate, Expr, ExprKind, FormatArgs};
 use rustc_data_structures::fx::FxHashMap;
 use rustc_lexer::{tokenize, TokenKind};
 use rustc_lint::{EarlyContext, EarlyLintPass};
-use rustc_session::{declare_tool_lint, impl_lint_pass};
+use rustc_session::impl_lint_pass;
 use rustc_span::{hygiene, Span};
 use std::iter::once;
 use std::mem;
 use std::rc::Rc;
 
-declare_clippy_lint! {
-    /// ### What it does
-    /// Collects [`rustc_ast::FormatArgs`] so that future late passes can call
-    /// [`clippy_utils::macros::find_format_args`]
-    pub FORMAT_ARGS_COLLECTOR,
```

Comment:
Why could this be removed?

Reply:
The lint is never actually used, and a lint pass doesn't need to have associated lints to be registered & run for side effects

I removed it because it was producing an error for the metadata collector which was previously ignored:

```
error: metadata collection error for `FORMAT_ARGS_COLLECTOR`: Unable to find lint in lint_store
  --> src/utils/format_args_collector.rs:19:9
   |
19 |     pub FORMAT_ARGS_COLLECTOR,
   |         ^^^^^^^^^^^^^^^^^^^^^
```